cmd/jag/commands/firmware.go

```diff
@@ -157,19 +158,46 @@ func BuildFirmwareEnvelope(ctx context.Context, id string, name string, wifiSSID
 		return nil, err
 	}
 
-	// TODO(kasper): Can we generate this in a nicer way?
-	wifiProperties := "{ \"wifi.ssid\": \"" + wifiSSID + "\", \"wifi.password\": \"" + wifiPassword + "\" }"
+	configMap := map[string]interface{}{
+		"id":   id,
+		"name": name,
+	}
+	configJson, err := json.Marshal(configMap)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := runFirmwareTool(ctx, sdk, envelopePath, "container", "install", "-o", envelope.Name(), "jaguar", jaguarSnapshot); err != nil {
+	configJsonFile, err := os.CreateTemp("", "*.json.assets")
+	if err != nil {
 		return nil, err
 	}
-	if err := setFirmwareProperty(ctx, sdk, envelope, "uuid", id); err != nil {
+	defer configJsonFile.Close()
+
+	if err := os.WriteFile(configJsonFile.Name(), configJson, 0666); err != nil {
+		return nil, err
+	}
+
+	assetsFile, err := os.CreateTemp("", "*.assets")
+	if err != nil {
+		return nil, err
+	}
+	defer assetsFile.Close()
+
+	if err := runAssetsTool(ctx, sdk, assetsFile.Name(), "create"); err != nil {
+		return nil, err
+	}
+
+	if err := runAssetsTool(ctx, sdk, assetsFile.Name(), "add", "--ubjson", "config", configJsonFile.Name()); err != nil {
 		return nil, err
 	}
-	if err := setFirmwareProperty(ctx, sdk, envelope, "id", id); err != nil {
+
+	// TODO(kasper): Can we generate this in a nicer way?
+	wifiProperties := "{ \"wifi.ssid\": \"" + wifiSSID + "\", \"wifi.password\": \"" + wifiPassword + "\" }"
```

Comment:
Might as well copy the approach from line 161.
However, that blows up the size from one line to 8 lines...